test/request-message.spec.js

```diff
@@ -0,0 +1,25 @@
+import {RequestMessage} from '../src/request-message';
+
+describe('RequestMessage', () => {
+  describe('fullUrl', () => {
+    var message;
+    beforeEach(() => {
+      message = new RequestMessage('get', '/anotherurl');
+      message.baseUrl = "example.com";
```

Comment:
Now that we have a base class, it might be nice to have it initialize its `baseUrl` property to the empty string, so things won't break if we fail to set it.

Also, please use single quotes for strings.